settings/resource_default_namespace_settings.go

```diff
@@ -0,0 +1,178 @@
+package settings
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/databricks/databricks-sdk-go/apierr"
+	"github.com/databricks/databricks-sdk-go/service/settings"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// NewDefaultNamespaceSettingsAPI creates a DefaultNamespaceSettingsAPI instance
+func NewDefaultNamespaceSettingsAPI(ctx context.Context, m any) DefaultNamespaceSettingsAPI {
+	client := m.(*common.DatabricksClient)
+	return DefaultNamespaceSettingsAPI{client, ctx}
+}
+
+// DefaultNamespaceSettingsAPI exposes the Default Namespace Settings API
+type DefaultNamespaceSettingsAPI struct {
+	client  *common.DatabricksClient
+	context context.Context
+}
+
+func (a DefaultNamespaceSettingsAPI) isEtagVersionError(err error) bool {
+	var aerr *apierr.APIError
+	if !errors.As(err, &aerr) {
+		return false
+	}
+	return aerr.StatusCode == http.StatusNotFound || (aerr.StatusCode == http.StatusConflict || aerr.ErrorCode == "RESOURCE_CONFLICT")
```

Comment:
We usually checking both for the 409 (conflict) case (see https://github.com/databricks/terraform-provider-databricks/pull/2371/files#diff-7eef2f7ec147f91e0b6597efd71eeece37d9dbf900e83f03d8f5f00d6011b7e7R76).

It may not be necessary, but it should not make a difference.

Reply:
> We usually checking both for the 409 (conflict) case (see https://github.com/databricks/terraform-provider-databricks/pull/2371/files#diff-7eef2f7ec147f91e0b6597efd71eeece37d9dbf900e83f03d8f5f00d6011b7e7R76).

That's an AND though - so it should be `(aerr.StatusCode == http.StatusConflict **&&** aerr.ErrorCode == "RESOURCE_CONFLICT")` instead of `(aerr.StatusCode == http.StatusConflict **||** aerr.ErrorCode == "RESOURCE_CONFLICT")`

Reply:
Updated